pennylane/fermi/fermionic.py

```diff
@@ -303,6 +303,132 @@ def to_mat(self, n_orbitals=None):
 
         return mat
 
+    def commute(self, source, target):
+        r"""Apply anti-commuting relations until the operator in position source is in position target.
+
+        Args:
+            source (int): The position of the operator to be commuted.
+            target (int): The desired position of the operator occupying the source position.
+
+        Returns:
+            FermiSentence: The ``FermiSentence`` obtained after applying the anti-commutator relations.
+
+        Raises:
+            TypeError: If the source or target is not an integer
+            ValueError: If the source or target are outside the range [0, len(self) - 1]
+
+
+        **Example**
+
+        >>> w = FermiWord({(0, 0): '+', (1, 1): '-'})
+        >>> w.commute(0, 1)
+        -1 * a(1) a⁺(0)
+        """
+
+        if not isinstance(source, int) or not isinstance(target, int):
+            raise TypeError("Indices must be integers")
+
+        if source < 0 or target < 0:
+            raise ValueError("Indices must be positive integers")
+
+        if source > len(self.sorted_dic) - 1 or target > len(self.sorted_dic) - 1:
+            raise ValueError("Indices out of range")
+
+        if source == target:
+            return FermiSentence({self: 1})
+
+        fw = self
```

Comment:
If we do `dict(self)`, we might be able to skip doing `dict(fw)` in each call of `_commute_adjacent`, right?

Reply:
You'd have to change ``_commute_adjacent`` to take a ``dict`` and return a ``dict`` instead of a ``FermiWord``. I think having the return type as ``(FermiSentence, FermiWord)`` is nicer than ``(FermiSentence, dict)``.